File: src/benchmark.js
const start = performance.now();

let array = new Array(10000).fill(0);

for (let i = 0; i < 10000; i++) {
  for (let j = 0; j < 100000; j++) {
    array[i] = array[i] + j;
  }
}

const end = performance.now();
console.log(`JavaScript execution time: ${(end - start).toFixed(2)} ms`); 

File: src/benchmark.go
package main

import (
	"fmt"
	"time"
)

func main() {
	const cacheLineSize = 64
	type paddedInt struct {
		value int
		pad   [cacheLineSize - 8]byte
	}

	start := time.Now()

	var array [10000]paddedInt

	for i := 0; i < 10000; i++ {
		for j := 0; j < 100000; j++ {
			array[i].value = array[i].value + j
		}
	}

	duration := time.Since(start)
	fmt.Printf("Go execution time: %.2f ms\n", float64(duration.Milliseconds()))
}
